src/components/Widgets/Form/AddressSelect.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
+import ReactSelect from 'react-select'
+import styled from 'styled-components'
+import Identicon from '../../Identicon'
+import { combineWallets } from '../../../util/helpers';
+
+export default class AddressSelect extends Component {
+  static displayName = 'AddressSelect';
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    wallets: PropTypes.object,
+    walletContracts: PropTypes.object,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    wallets: {},
+    walletContracts: {},
+  };
+
+  state = {
+    selectedWallet: '',
+    combinedWallets: [],
+  }
+
+  componentDidMount() {
+    const { wallets, walletContracts, onChange } = this.props;
+    const combinedWallets = combineWallets(wallets, walletContracts);
+    let wallet = combinedWallets[0]
+    const address = wallet ? wallet.address : ''
+    this.setState(
+      { combinedWallets, selectedWallet: {value: address, ...wallet} },
+      onChange(combinedWallets[0])
+    )
+  }
+
+  selectAddress = e => {
+    const { onChange } = this.props;
+    this.setState({ selectedWallet: e }, onChange(e))
+  }
+
+  render() {
+    const { disabled } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+
+    let options = combinedWallets.map(w => {
+      return {
+        value: w.address,
+        label :w.addressType === 'wallet' 
```

Comment:
You've still got formatting and linting issues still hanging around. Please, please pause to integrate `prettier` into your editor (https://prettier.io/docs/en/editors.html) and use the linter in your editor or run `yarn lint:watch` in your terminal to spot lints in files you've worked in. Also, looks like you didn't actually rebase or merge master into your work, so recreating the yarn.lock file didn't have the desired effect of resolving conflicts.